code/game/objects/effects/portals.dm

```diff
@@ -124,12 +127,18 @@
 		linked = null
 	return ..()
 
+/obj/effect/portal/attack_ghost(mob/dead/observer/O)
+	if(!teleport(O))
+		return ..()
+
 /obj/effect/portal/proc/teleport(atom/movable/M)
-	if(!istype(M) || istype(M, /obj/effect) || isobserver(M) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M))) //Things that shouldn't teleport.
+	if(!istype(M) || istype(M, /obj/effect) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M))) //Things that shouldn't teleport.
 		return
 	var/turf/real_target = get_link_target_turf()
 	if(!istype(real_target))
 		return FALSE
+	if(M.anchored && !allow_anchored)

```

Comment:
mechs are anchooooored. i swear people keep trying to do this, but mechs are anchored.

Reply:
fixing soon!